### ISSUE TITLE
Functions and modules cannot share the same name

### ISSUE THREAD

**Issue by EmilyBourne** [pyccel/pyccel#344]
The following file will not compile as the fortran module receives the same name as the function.

fib.py:
```
from pyccel.decorators import types

@types('int', results='int')
def fib(n) :
    if n < 2:
        result = n
        return result
    result = fib (n - 1) + fib (n - 2)
    return result
```

The problem could be fixed by adding a prefix to the module name however name clashes cannot be definitively eliminated with this method as the module name must be deterministic in order to be identified when used for imports.
A new module name could be defined deterministically at the semantic stage as dependencies are fully parsed and annotated however this may not always be the case in the future.

Even if we cannot fully avoid all name clashes I think a module prefix could still be useful as giving the file and the function the same name is relatively common

**Comment by yguclu** [comment]
Maybe duplicated: see issue #227

**Comment by yguclu** [comment]
Information about the generated names could be stored in the `.pyh` file; this would avoid the requirement of generating deterministic names.

**Comment by EmilyBourne** [comment]
Fixed by #1089
